### PR TITLE
[Woo POS] Remove POS access reliance on beta feature toggle switch

### LINKED ISSUE #13838: [Woo POS] Improvement: Remove experimental toggle flag
From p1725337084101939-slack-C070SJRA8DP

> Now that the remote feature flag is accessible . Should we also remove the "experimental feature" POS toggle from Menu > Settings for simplicity? This was added so we could start to work on the POS right away, but since we can whitelist specific accounts we could just rely on this + a general feature flag set to localDev/alpha

### PR DESCRIPTION
**Pull request by iamgabrielma**
Ref: p1725337084101939-slack-C070SJRA8DP

https://github.com/woocommerce/woocommerce-ios/pull/13815 must be merged first.
Closes: #13838

## Description
This PR removes the toggle switch from `Menu > Experimental Features > Point of Sale`, so rendering the point of entry to access POS only relies on the `POSEligibilityChecker.isEligible` publisher now, rather than both the toggle + the publisher. The POS row will only appear on developer builds, as long as the store is eligible for POS.

There are no changes in `isEligible`, since this already checks for `displayPointOfSaleToggle`, which is set to only return true on `buildConfig == .localDeveloper || buildConfig == .alpha` 

## Changes
* Removes reliance on the beta feature flag toggle and related AppSettingsAction / AppSettingsStore code
* Updates feature flag name

## Testing:
1. On an eligible store for POS (WooCommerce 6.6+, IPP onboarding completed, whitelisted in `mobile/feature-flags/`, and using an iPad):
2. Run the app in `Debug` build configuration:
Screenshot 2024-09-04 at 13 19 06

3. Observe there is no toggle under `Menu > Experimental Features > Point of Sale`
4. Observe that the POS row is visible

Simulator Screenshot - iPad Air 11 - iOS 17 5 M2 - 2024-09-04 at 14 19 21

5. Repeat for `Release` build configuration
6. Observe there still is no toggle under `Menu > Experimental Features > Point of Sale`
7. Observe that the POS row is no longer visible

Simulator Screenshot - iPad Air 11 - iOS 17 5 - 2024-09-04 at 14 26 26

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by staskus** [comment]
> The POS row will only appear on developer builds, as long as the store is eligible for POS.

I don't think we should do that.

POS row **should** appear on release builds as well if the store is eligible for POS.

Therefore, the logic should be:

- Visible = debug || alpha || remote_flag

Instead of:

- Visible = (debug || alpha) && remote_flag

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/POS/POSEligibilityChecker.swift [hunk 1]
```diff
@@ -19,7 +19,7 @@ final class POSEligibilityChecker: POSEligibilityCheckerProtocol {
     var isEligible: AnyPublisher<Bool, Never> {
         // Conditions that are fixed for its lifetime.
         let isTablet = userInterfaceIdiom == .pad
-        let isFeatureFlagEnabled = featureFlagService.isFeatureFlagEnabled(.displayPointOfSaleToggle)
+        let isFeatureFlagEnabled = featureFlagService.isFeatureFlagEnabled(.pointOfSale)
```
Comment (staskus) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/POS/POSEligibilityChecker.swift]: My suggestion is to use this `featureFlagService.isFeatureFlagEnabled(.pointOfSale)` together with remoteFeatureFlag to determine visibility. 

Or, if interested, implement some logic within the remote feature flag, so its value could be conditionally overridden (https://github.com/wordpress-mobile/WordPress-iOS/blob/add21ecd1286de56f85650a4443c8de18a9f309b/WordPress/Classes/Utility/BuildInformation/RemoteFeatureFlag.swift#L242). First, in the code, later if we need it, in some debug menu as well.

Comment (iamgabrielma) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/POS/POSEligibilityChecker.swift]: > My suggestion is to use this featureFlagService.isFeatureFlagEnabled(.pointOfSale) together with remoteFeatureFlag to determine visibility.

I updated the logic on https://github.com/woocommerce/woocommerce-ios/pull/13839/commits/7f27a6096a49aa669ecd83a31c48bec1ee6e1724 so we only have a single `isPointOfSaleFeatureFlagEnabled` publisher, which checks both remote and local flags. Let me know what you think:

1. If the account is whitelisted -> The remote feature flag returns true -> `isPointOfSaleFeatureFlagEnabled` always returns true for any build.
2. If the account is not-whitelisted ->
	2.1 Developer build -> `isPointOfSaleFeatureFlagEnabled` returns true, row will show.
	2.1 Release build -> `isPointOfSaleFeatureFlagEnabled` returns false, row won't show.

**Comment by iamgabrielma** [comment]
> I don't think we should do that.
> POS row should appear on release builds as well if the store is eligible for POS.

That sounds better indeed, I was a bit concerned about dev/testing with the remote flag returning false for non-Jetpack connected sites and the limitation of simulators with a8c accounts, but this change makes it straight-forward again for dev builds and release builds without having to worry about the account we use.

**Review by staskus — APPROVED** [review]
Thank you for the fixes!

Tested:
- Debug build & illegible user = POS option appears
- Release build & illegible user = No POS option
- Release build & supported user = POS option appears

**Pull request merged**